lib/_http_outgoing.js

```diff
@@ -79,7 +79,7 @@ exports.OutgoingMessage = OutgoingMessage;
 
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
-  if (callback)
+  if (typeof callback === 'function')
```

Comment:
Agreed. I was attempting to avoid adding a new throw but you're right, it's better to be explicit about the error here